jax/interpreters/pxla.py

```diff
@@ -297,30 +279,38 @@ class ShardedDeviceArray(xla.DeviceArray):
   behavior of an ndarray so that it can be treated by user code as an ndarray;
   that is, it is only an optimization to reduce transfers.
 
-  The number of device buffers underlying a ShardedDeviceArray instance is equal
-  to the number of replicas of the computation that produced it. Each buffer
-  represents a shard of the original array, meaning a slice along its leading
-  axis. These component buffers reside on distinct devices, but need not
-  represent distinct logical shards. The correspondence can be computed with
-  the assign_shards_to_replicas function.
+  Each device buffer resides on a distinct device, and has a corresponding index
```

Comment:
There are important constraints that are mentioned in the design-doc but not here: the shards are exhaustive, non-overlapping, same-size. Entire shards may be replicated on multiple devices. 

I also think that we should either have assertions about these on construction, or (better) a different representation of the shard that is higher-level and which ensures these constraints automatically.